fmn/rules/generic.py

```diff
@@ -143,7 +151,11 @@ def user_package_watch_filter(config, message, fasnick=None, *args, **kw):
             return False
         usr_packages = fmn.rules.utils.get_packages_of_user(
             config, fasnick, only_watchcommits)
-        return usr_packages.intersection(msg_packages)
+        for namespace, packages in usr_packages.items():
+            namespaced_msg_packages = fmn.rules.utils.msg2packages(
+                message, namespace=namespace, **config)
+            if packages.intersection(namespaced_msg_packages):
+                return True
```

Comment:
This code block seem duplicated with the other two changes in this file. Perhaps make a shared function?